extensions/HuggingFace/python/src/aiconfig_extension_hugging_face/local_inference/text_summarization.py

```diff
@@ -243,16 +245,15 @@ async def run_inference(self, prompt: Prompt, aiconfig: "AIConfigRuntime", optio
         completion_data = await self.deserialize(prompt, aiconfig, options, parameters)
         inputs = completion_data.pop("prompt", None)
 
-        model_name: str = aiconfig.get_model_name(prompt)
-        if isinstance(model_name, str) and model_name not in self.summarizers:
-            self.summarizers[model_name] = pipeline("summarization", model=model_name)
-        summarizer = self.summarizers[model_name]
+        model_name = get_hf_model(aiconfig, prompt, self)
+        key = model_name if model_name is not None else "__default__"
+        if key not in self.summarizers:
+            self.summarizers[key] = pipeline("summarization", model=model_name)
+        summarizer = self.summarizers[key]
 
         # if stream enabled in runtime options and config, then stream. Otherwise don't stream.
         streamer = None
-        should_stream = (options.stream if options else False) and (
-            not "stream" in completion_data or completion_data.get("stream") != False
-        )
+        should_stream = (options.stream if options else False) and (not "stream" in completion_data or completion_data.get("stream") != False)
```

Comment:
not for this diff but this stream logic should be cleaned up

Reply:
+1

Reply:
I have an bootcamp issue to track this in https://github.com/lastmile-ai/aiconfig/issues/861